lib/components/Attachments/index.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useRef, useEffect, useState } from "react";
+
+import classnames from "classnames";
+import { Button, Typography, Toastr } from "neetoui";
+
+import { DIRECT_UPLOAD_ENDPOINT } from "common/constants";
+import useUppyUploader from "hooks/useUppyUploader";
+
+import Attachment from "./Attachment";
+import AttachmentProgress from "./AttachmentProgress";
+import { DEFAULT_UPPY_CONFIG } from "./constants";
+
+const Attachments = ({
+  endpoint = DIRECT_UPLOAD_ENDPOINT,
+  attachments = [],
+  className,
+  onChange,
+}) => {
+  const [pendingAttachments, setPendingAttachments] = useState([]);
+
+  const addAttachmentRef = useRef(null);
+
+  const { uppy, isUploading } = useUppyUploader({
+    endpoint,
+    uppyConfig: DEFAULT_UPPY_CONFIG,
+  });
+
+  const handleAddFile = event => {
+    const files = Array.from(event.target.files);
+    files.forEach(file => {
+      uppy.addFile({
+        name: file.name,
+        type: file.type,
+        data: file,
+      });
+    });
+
+    const newlyAddedFiles = files.map(file => ({
+      filename: file.name,
+      signedId: "awaiting",
+      url: "",
+      progress: 0,
+    }));
+    setPendingAttachments(prevState => [...prevState, ...newlyAddedFiles]);
```

Comment:
Please let me know if this is the preferable way to add files to pending state or should I make changes to the mentioned code?

Reply:
LGTM

Reply:
Okay.